recipes/serd/all/conanfile.py

```diff
@@ -47,46 +59,77 @@ def configure(self):
             del self.options.fPIC
 
     def validate(self):
-        if tools.cross_building(self):
-            raise ConanInvalidConfiguration("Cross compiling is not supported by serd's build system Waf.")
+        if Version(self.version) > Version("0.30.12"):
+            if cross_building(self):
+                raise ConanInvalidConfiguration("Cross compiling is not working.")
+            if is_msvc(self):
+                raise ConanInvalidConfiguration("Meson packaging is broken for MSVC.")
+        else:
+            if self.settings.compiler == "apple-clang" and Version(self.settings.compiler.version) < 12:
+                raise ConanInvalidConfiguration("apple-clang < 12 is not supported")
+
+            if cross_building(self):
+                raise ConanInvalidConfiguration("Cross compiling is not supported by serd's build system Waf.")
 
-        if is_msvc(self):
-            raise ConanInvalidConfiguration("Don't know how to setup WAF for VS.")
+            if is_msvc(self):
+                raise ConanInvalidConfiguration("Don't know how to setup WAF for VS.")
+
+    def _configure_meson(self):
+        if self._meson:
+            return self._meson
+        self._meson = Meson(self)
+        args = ["--wrap-mode=nofallback"]
+        defs = {"docs": "disabled", "tests": "disabled", "tools": "disabled"}
+        self._meson.configure(source_folder=self.folders.source_folder,
+                              build_folder=os.path.join(self.package_folder, "build"),
+                              args=args, defs=defs)
+        return self._meson
 
     def build(self):
-        args = ["--no-utils", " --prefix={}".format(self.folders.package_folder)]
-        if not self.options.shared:
-            args += ["--static", "--no-shared"]
-        args = " ".join(arg for arg in args)
-
-        cflags = []
-        if self.options.get_safe("fPIC"):
-            cflags += ["-fPIC"]
-        if self.settings.build_type in ["Debug", "RelWithDebInfo"]:
-            cflags += ["-g"]
-        if self.settings.build_type in ["Release", "RelWithDebInfo"]:
-            cflags += ["-O3"]
-        if self.settings.build_type in ["Release", "MinSizeRel"]:
-            cflags += ["-DNDEBUG"]
-        if self.settings.build_type == "MinSizeRel":
-            cflags += ["-Os"]
-        cflags = " ".join(cflag for cflag in cflags)
-
-        self.run(f'CFLAGS="{cflags}" ./waf configure {args}', run_environment=True)
-        self.run('./waf build', run_environment=True)
+        if Version(self.version) > Version("0.30.12"):
```

Comment:
Given how complicate the old code... maybe we can drop the older version?

CCI will always make the older one available but it would simply this recipe (and hopefully they wont change back lol)

Reply:
I would be happy to remove it.  :) I am quite sure they won't change back. I think the meson build system is a backport from the upcoming 1.x.x version.

This includes:
- remove all code for 0.30.12 from serd/all/conanfile.py
  - would I keep 
    ```
    if Version(self.version) > Version("0.30.12"):
    ```
    in validate and throw a `ConanInvalidConfiguration` instead? If yes, what would be the exception message?
- comment out the versions in serd/all/conandata.yml
- comment out the versions in serd/config.yml

correct?

Reply:
You can always delete the lines from conandata and config (but commenting is fine, I've seen it in other recipes)

I would get ride of the conditon all together... you should not need to throw exceptions for the older version.
This recipe just wont know something older ever existed